### ISSUE TITLE
StorageError with other packages

### ISSUE THREAD

**Issue by kowsheek** [Azure/azure-storage-node#247]
I am using the blob client to upload files to Azure storage. It's been fine but I've been updating the packages across my project and the upload is failing with the error below. Seems to me that the error is coming from the library and not from Azure (since I didn't see a request go out) while checking if the container exists.

Having played around with the packages, it seems like one of the other packages I'm using is making this library act up. I triple checked the access keys and they are matched up.

**Environment**

NodeJS: v6.9.3 (tried with v7.3.0 as well)

```
"dependencies": {
    "applicationinsights": "^0.17.1",
    "azure-storage": "^1.4.0",
    "body-parser": "~1.15.1",
    "busboy": "^0.2.13",
    "composable-middleware": "^0.3.0",
    "compression": "~1.6.2",
    "connect-mongodb-session": "^1.3.0",
    "cookie-parser": "~1.4.2",
    "ejs": "^2.5.5",
    "errorhandler": "^1.5.0",
    "express": "^4.14.0",
    "express-jwt": "^5.1.0",
    "express-session": "^1.14.2",
    "humanparser": "^1.1.1",
    "jsonwebtoken": "^7.2.1",
    "lodash": "^4.17.4",
    "method-override": "^2.3.7",
    "moment": "^2.17.1",
    "mongoose": "^4.7.6",
    "morgan": "~1.7.0",
    "passport": "~0.3.2",
    "passport-local": "~1.0.0",
    "q": "^1.4.1",
    "sendgrid": "^4.7.1",
    "serve-favicon": "^2.3.2",
    "socket.io": "^1.7.2",
    "socket.io-client": "^1.7.2",
    "socketio-jwt": "^4.3.4",
    "stripe": "^4.14.0",
    "twilio": "^2.11.1"
  }
```

**Actual**

Receiving a 403 from the library and no requests are being made to Azure (monitored by Fiddler)

`{"name":"StorageError","message":"Forbidden","code":"Forbidden","statusCode":403,"requestId":"21bb0273-0001-0069-75cc-66a5b2000000"},"status":500,"config":{"method":"POST","transformRequest":[null],"transformResponse":[null],"jsonpCallbackParam":"callback","url":"api/attachments/upload","data":{"file":{}},"_isDigested":true,"_chunkSize":null,"headers":{"Accept":"application/json, text/plain, */*","Authorization":"Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI1ODZkNjA1Nzk1MTg3ZTJlNjA0OWEyNjciLCJpYXQiOjE0ODM1NjMxNTUsImV4cCI6MTQ4NDg1OTE1NX0.UFTiNn8yni3vYyeStyCTxDduK6FRESx0owI2EK9V45U"},"_deferred":{"promise":{}}},"statusText":"Internal Server Error"}`

**Exptected**

A 200 and upload of the file to the Azure storage account.

**Comment by kowsheek** [comment]
Seems like [`"applicationinsights": "^0.17.1",`](https://github.com/Microsoft/ApplicationInsights-node.js/releases/tag/v0.17.1) is the culprit. Having it in v0.15.16 works with this library at v1.4.0.

**Comment by hasonmsft** [comment]
@kowsheek , applicationinsights 0.17.1 will always inject some telemetry header globally before sending the request. While it's done after the HMAC calculation (for authentication), it'll break the authentication as the HMAC will mismatch between client and the service.

**Comment by stefan-akelius** [comment]
Can confirm this error. Using the `azure-storage@1.4.0` and `applicationinsights@0.17.1` packages in an Azure Function blocks all calls to any Azure Storage client.

**Comment by kowsheek** [comment]
@hasonmsft What do you propose as a solution?

**Comment by kamilszostak** [comment]
This is a known bug in `applicationinsights@0.17.1`, which will be addressed in next release (v0.18.0). For now please use `applicationinsights@0.16.0`.

**Comment by kowsheek** [comment]
Thanks @kamilszostak!
